hw2/src/main/java/edu/csula/storage/servlet/UsersDAOImpl.java

```diff
@@ -0,0 +1,51 @@
+package edu.csula.storage.servlet;
+
+import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Optional;
+
+import javax.servlet.http.HttpSession;
+
+import edu.csula.models.User;
+import edu.csula.storage.UsersDAO;
+
+/**
+ * To abstract the storage access from the business layer using HttpSession
+ */
+public class UsersDAOImpl implements UsersDAO {
+	private final HttpSession context;
+	protected static final String CONTEXT_NAME = "users";
+
+	public UsersDAOImpl(HttpSession context) {
+		this.context = context;
+	}
+
+	@Override
+	public boolean authenticate(String username, String password) {
+		// TODO: check if username/password combination is valid and store the
+		//       username/password into the session
+		User user = new User(0, username, password);
+
+		context.setAttribute(CONTEXT_NAME, user);
```

Comment:
Should not set attribute when the authentication fails. e.g. if the user login with wrong credential, the getAuthenticatedUser would still return true.